src/libraries/System.Private.CoreLib/src/System/IO/BufferedStream.cs

```diff
@@ -403,12 +487,14 @@ private void ClearReadBufferBeforeWrite()
             // However, since the user did not call a method that is intuitively expected to seek, a better message is in order.
             // Ideally, we would throw an InvalidOperation here, but for backward compat we have to stick with NotSupported.
             if (!_stream.CanSeek)
-                throw new NotSupportedException(SR.NotSupported_CannotWriteToBufferedStreamIfReadBufferCannotBeFlushed);
+                Throw();
 
             FlushRead();
+
+            static void Throw() => throw new NotSupportedException(SR.NotSupported_CannotWriteToBufferedStreamIfReadBufferCannotBeFlushed);
         }
 
-        private void FlushWrite()
+        private void FlushWrite(bool performActualFlush)
```

Comment:
Who passes in false, and why?

Reply:
The only case I can find where this boolean could potentially be `false` is when coming from the callstack:

- `BufferedStream.FlushWrite(bool)`
- `BufferedStream.Flush(bool)`
- `BufferedFileStreamStrategy.Flush(bool)`
- Either `FileStream.Flush(bool)` or  `DerivedFileStreamStrategy.Flush(bool)`

I agree that an explanation of when this argument can be `false` would be helpful, above all the methods involved in the callstacks that could pass `false`.

Reply:
I added a comment